File: backend/internal/delivery/dto/http_dto.go
package dto

// CreateGameRequest represents the request body for creating a game
type CreateGameRequest struct {
	MaxPlayers int `json:"maxPlayers" binding:"required,min=1,max=5"`
}

// CreateGameResponse represents the response for creating a game
type CreateGameResponse struct {
	Game GameDto `json:"game" ts:"GameDto"`
}

// JoinGameRequest represents the request body for joining a game
type JoinGameRequest struct {
	PlayerName string `json:"playerName" binding:"required,min=1,max=50"`
}

// JoinGameResponse represents the response for joining a game
type JoinGameResponse struct {
	Game     GameDto `json:"game" ts:"GameDto"`
	PlayerID string  `json:"playerId" ts:"string"`
}

// GetGameResponse represents the response for getting a game
type GetGameResponse struct {
	Game GameDto `json:"game" ts:"GameDto"`
}

// ListGamesResponse represents the response for listing games
type ListGamesResponse struct {
	Games []GameDto `json:"games" ts:"GameDto[]"`
}

// UpdateResourcesRequest represents the request body for updating player resources
type UpdateResourcesRequest struct {
	Resources ResourcesDto `json:"resources" ts:"ResourcesDto"`
}

// GetPlayerResponse represents the response for getting a player
type GetPlayerResponse struct {
	Player PlayerDto `json:"player" ts:"PlayerDto"`
}

// UpdatePlayerResourcesResponse represents the response for updating player resources
type UpdatePlayerResourcesResponse struct {
	Player PlayerDto `json:"player" ts:"PlayerDto"`
}

// ListCardsResponse represents the response for listing cards with pagination
type ListCardsResponse struct {
	Cards      []CardDto `json:"cards" ts:"CardDto[]"`
	TotalCount int       `json:"totalCount" ts:"number"`
	Offset     int       `json:"offset" ts:"number"`
	Limit      int       `json:"limit" ts:"number"`
}

// ErrorResponse represents an error response
type ErrorResponse struct {
	Error   string `json:"error" ts:"string"`
	Code    string `json:"code,omitempty" ts:"string"`
	Details string `json:"details,omitempty" ts:"string"`
}

File: backend/src/main/kotlin/com/terraformingmars/dto/HttpDto.kt
package com.terraformingmars.dto

import kotlinx.serialization.Serializable

// HTTP Request/Response DTOs matching the Go DTOs

@Serializable
data class CreateGameRequest(
    val maxPlayers: Int,
    val playerName: String
)

@Serializable
data class CreateGameResponse(
    val game: GameDto
)

@Serializable
data class JoinGameRequest(
    val playerName: String
)

@Serializable
data class JoinGameResponse(
    val game: GameDto,
    val playerId: String
)

@Serializable
data class GetGameResponse(
    val game: GameDto
)

@Serializable
data class ListGamesResponse(
    val games: List<GameDto>
)

@Serializable
data class UpdateResourcesRequest(
    val resources: ResourcesDto
)

@Serializable
data class GetPlayerResponse(
    val player: PlayerDto
)

@Serializable
data class UpdatePlayerResourcesResponse(
    val player: PlayerDto
)

@Serializable
data class ListCardsResponse(
    val cards: List<CardDto>,
    val totalCount: Int,
    val offset: Int,
    val limit: Int
)

@Serializable
data class ErrorResponse(
    val error: String,
    val code: String? = null,
    val details: String? = null
)